.github/actions/observe-build-status/action.yml

```diff
@@ -75,30 +81,33 @@ runs:
         else
           echo "result=" >> $GITHUB_OUTPUT
         fi
-
-    ####################### DETAILED FLAKY JUNIT TESTS #########################
-    - name: Get detailed info on flaky JUnit tests
-      id: get-detailed-flaky-junit-tests
-      if: ${{ inputs.detailed_junit_flaky_tests == 'true' && steps.secrets.outputs.gcloud_sa_key != '' }}
+    ######################## Report test result #############################  
+    - name: Get detailed info on JUnit tests
+      id: get-detailed-junit-tests
+      if: ${{ (inputs.detailed_junit_tests == 'true' || inputs.detailed_junit_flaky_tests == 'true') && steps.secrets.outputs.gcloud_sa_key != '' }}
       shell: bash
-      # To support multi-line string in output we have to work with EOF delimiter
-      # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-a-multiline-string
       run: |
-        {
-          echo 'result<<EOF'
-          find . -iname 'TEST-*-FLAKY.xml' | xargs -d '\n' --no-run-if-empty -n1 python3 .github/actions/observe-build-status/junit-flaky-to-jsonl.py
-          (find . -iname 'TEST-*.xml' -exec grep -l '</failure>' {} + | grep -v 'FLAKY.xml' | xargs -d '\n' --no-run-if-empty -n1 python3 .github/actions/observe-build-status/junit-failure-to-jsonl.py || true)
-          echo 'EOF'
-        } >> $GITHUB_OUTPUT
+        TEST_RESULTS_FILE=$(mktemp -t junit_test_results.${GITHUB_JOB}_${GITHUB_RUN_ID}_${GITHUB_RUN_ATTEMPT}.XXXXXX.jsonl)
```

Comment:
ℹ️ Each GHA job gets an isolated runner with a fresh filesystem, so there is technically no need to include environment variables in the temporary file name here - there can't be name collisions with other jobs as they have their own runners.